exec/utils.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024-Present Defense Unicorns
+
+package exec
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+	"sync"
+
+	"github.com/defenseunicorns/pkg/helpers"
+)
+
+var supportedCmdMutations = []string{"zarf", "uds", "kubectl"}
+var registeredCmdMutations = sync.Map{}
+
+// GetFinalExecutablePath returns the absolute path to the current executable, following any symlinks along the way.
+func GetFinalExecutablePath() (string, error) {
+	binaryPath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	// In case the binary is symlinked somewhere else, get the final destination
+	return filepath.EvalSymlinks(binaryPath)
+}
+
+// RegisterCmdMutation registers local ./ commands that should change to the specified cmdLocation
+func RegisterCmdMutation(cmdKey string, cmdLocation string) error {
+	if slices.Contains(supportedCmdMutations, cmdKey) {
+		registeredCmdMutations.Store(cmdKey, cmdLocation)
+		return nil
+	}
+	return fmt.Errorf("%s is not a supported command key", cmdKey)
+}
+
+// GetCmdMutation returns the cmdLocation for a given cmdKey and whether that key exists
+func GetCmdMutation(cmdKey string) (string, error) {
+	if slices.Contains(supportedCmdMutations, cmdKey) {
+		cmdLocation, ok := registeredCmdMutations.Load(cmdKey)
+		if !ok {
+			cmdLocation = cmdKey
+		}
+		return cmdLocation.(string), nil
+	}
+	return "", fmt.Errorf("%s is not a supported command key", cmdKey)
+}
```

Comment:
```suggestion
// GetCmdMutation returns the cmdLocation for a given cmdKey and whether that key exists
func GetCmdMutation(cmdKey string) (string, bool) {
	if !slices.Contains(supportedCmdMutations, cmdKey) {
		return "", false
	}
	value, ok := registeredCmdMutations.Load(cmdKey)
	if !ok {
		return "", false
	}
	cmdLocation, ok := cmdLocation.(string)
	if !ok {
		return "", false
	}
	return cmdLocation, true
}
```

Leave it up to the caller to decide what to do when the key they want does not exist.

This also re-orients the code to keep the golden path as least-indented as possible.

Reply:
Hmm - wondering what the use case would be for not replacing?  If an app wanted to disable these they could register zarf to ./zarf and have it be turned off - having a default makes this easier to use for apps that don't have a location (since not all would).